### PR TITLE
Implements multiple UnitClass functions.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

**Review by OmniBlade — CHANGES_REQUESTED** [review]
You need newlines in UnitClass::Greatest_Threat and there are a couple of returns that follow a scope that need a new line before them in other functions.

File: src/game/engine/techno.cpp [hunk 1]
```diff
@@ -1633,20 +1633,45 @@ VisualType TechnoClass::Visual_Character(BOOL flag) const
...
 
 /**
- *
+ * Kill cargo, record the kill for the killer
  *
  */
 void TechnoClass::Kill_Cargo(TechnoClass *object)
 {
     while (m_Cargo.Has_Cargo()) {
-        FootClass *ptr = m_Cargo.Detach_Object();
```
Comment (OmniBlade) [inline src/game/engine/techno.cpp:1642]: Its unnessesary to split the assignment from the declaration of the variable in the code as it currently stands, please revert.

File: src/game/engine/techno.cpp [hunk 2]
```diff
@@ -1633,20 +1633,45 @@ VisualType TechnoClass::Visual_Character(BOOL flag) const
...
+        /*
+        // TS enhancement, likely to prevent a crash
+        // TODO commented out for now, investigate if this doesn't have side effects.
+        ptr = Attached_Object();
+
+        if (ptr->Get_Team() != nullptr) {
+            ptr->Get_Team()->Remove(ptr);
+        }
+        */
+        ptr = m_Cargo.Detach_Object();
```
Comment (OmniBlade) [inline src/game/engine/techno.cpp]: Newline.

File: src/game/engine/unit.cpp [hunk 3]
```diff
@@ -41,6 +96,20 @@ UnitClass::UnitClass(const NoInitClass &noinit) :
...
+            m_Team = nullptr;
+        }
+
+        m_OwnerHouse->Tracking_Remove(this);
+
+        Destory_Cargo();
+
+        Limbo();
+    }
+    m_Class = nullptr;
```
Comment (OmniBlade) [inline src/game/engine/unit.cpp:97]: Newline.

File: src/game/engine/unit.cpp [hunk 4]
```diff
@@ -288,14 +422,35 @@ DirType UnitClass::Desired_Load_Dir(ObjectClass *object, cell_t &cellnum) const
...
-#ifdef GAME_DLL
-    DEFINE_CALL(func, 0x005803C4, DirType, const UnitClass *);
-    return func(this);
-#else
-    return DIR_NONE;
-#endif
+    if (Class_Of().Is_Turret_Equipped()) {
+        // Special logics for the V2 facing.
+        if (What_Type() == UNIT_V2RL) {
+            // TODO figure out what exactly is going on here and document it.
```
Comment (OmniBlade) [inline src/game/engine/unit.cpp]: Its working out which is the minimum distance needed to turn by trying both directions.

File: src/game/engine/unit.cpp [hunk 5]
```diff
@@ -436,7 +694,7 @@ BOOL UnitClass::Ok_To_Move(DirType dir)
     if (m_Rotating) {
         return false;
     }
-    if (dir - m_TurretFacing.Get_Current() != 0) {
+    if (m_TurretFacing.Difference_From_Current(dir)) {
```
Comment (OmniBlade) [inline src/game/engine/unit.cpp]: Should still check != 0 as its not a boolean value being checked. Be explicit so its clear at a glance and to less experienced programmers what you meant. Also, needs newline between scopes.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**